src/main/java/org/testng/internal/Parameters.java

```diff
@@ -191,12 +192,12 @@ private static void checkParameterTypes(String methodName,
     }
 
     if (parameterNames.length != totalLength) {
-      throw new TestNGException( "Method " + methodName + " requires "
-          + parameterTypes.length + " parameters but "
-          + parameterNames.length
-          + " were supplied in the "
-          + methodAnnotation
-          + " annotation.");
+      throw new TestNGException(
```

Comment:
I'm not sure this exception is related to "if (parameterNames.length != totalLength)".
I think we can have more context somewhere.